### PR TITLE
(refactor) lib/db.js into es6 classes

### PR DESCRIPTION
**Pull request by jniles**
This commit rewrites the db modules as es6 classes.  The module is API-compatible with the previous version of `db`.  The transaction logic has been broken out into a separate class `Transaction` for testability. The files `db/index.js` and `db/transaction.js` now replace `db.js`.

There is no longer any `db.initialise()` method.  The first `require() will initialise a database connection from environmental variables.

The documentation has improved drastically.  Each method is now well documented using JSDoc standards.

Finally, nicer errors have been implemented for API/integration tests.  This makes the testing library log the HTTP method and URL in case programmers forget to accurately describe their assertions.

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -163,7 +161,7 @@ gulp.task('client-compile-vendor', function () {
 });
 
 
-// minify the client css styles via minifycss
+// minify the client css styles via cssnano
```
Comment (sfount) [inline gulpfile.js:153]: 😆

**Comment by jniles** [comment]
@sfount, can I get a review?

**Comment by jniles** [comment]
@sfount, can I get a review?

**Comment by sfount** [comment]
This is a huge improvement to the database connector. It is worrying that there are no unit tests to verify that these changes provide the same functionality however the integration tests heavily relying on `db` go a long way. 

In the future we could either look into an additional library or write some tests to cover this functionality.

**Pull request merged**